core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -125,6 +131,16 @@ TableScanContext shouldReturnColumnStats(boolean returnColumnStats) {
         .build();
   }
 
+  TableScanContext shouldReturnSpecificColumnStats(Collection<Integer> columStatsToInclude) {
```

Comment:
`should` is typically used for boolean arg. how about just `columnsToIncludeStats(Set<Integer> columns)`? similar to the `selectColumns` method below.